komodor/users.go

```diff
@@ -0,0 +1,116 @@
+package komodor
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/samber/lo"
+)
+
+const UsersUrl string = V2Endpoint + "/users"
+
+type User struct {
+	Id          string `json:"id"`
+	DisplayName string `json:"displayName"`
+	Email       string `json:"email"`
+	CreatedAt   string `json:"createdAt"`
+	UpdatedAt   string `json:"updatedAt"`
+}
+
+type NewUser struct {
+	DisplayName      string `json:"displayName"`
+	Email            string `json:"email"`
+	RestoreIfDeleted bool   `json:"restoreIfDeleted"`
+}
+
+type UpdateUser struct {
+	DisplayName string `json:"displayName"`
+}
+
+type getUsersParams struct {
+	IsDeleted *bool `json:"isDeleted,omitempty"`
+}
+
+func (c *Client) GetUsers() ([]User, error) {
+	return c.getUsers(&getUsersParams{IsDeleted: lo.ToPtr(false)})
```

Comment:
Shouldn't this `isDeleted` param be configurable in case they want to recover a user?
I do know that calling CreateUser with the same email should work, but do we have no use of listing the deleted ones at all?

Reply:
Most likely to the regular Terraform user that uses this I don't think there's a use for it, I was just wondering if I missed something maybe :)